echemdb/data/cv/entry.py

```diff
@@ -195,177 +197,132 @@ def format_title(self, e, which_field, as_sentence=True):
+        A rescaled entry with updated axes units::
 
-            >>> entry.y_unit(yunit='original')
-            Unit("mA / cm2")
+            >>> rescaled_entry = entry.rescale({'j':'uA / cm2', 't':'h'})
+            >>> rescaled_entry.data_description.fields # doctest: +NORMALIZE_WHITESPACE
+            [{'name': 't', 'unit': 'h', 'type': 'number', 'format': 'default'},
+            {'name': 'E', 'unit': 'V', 'reference': 'RHE', 'type': 'number', 'format': 'default'},
+            {'name': 'j', 'unit': 'uA / cm2', 'type': 'number', 'format': 'default'}]
 
-        Units can be specified explicitly::
-
-            >>> entry = Entry.create_examples()[0]
-            >>> entry.y_unit(yunit='uA / cm2')
-            Unit("uA / cm2")
+        A rescaled dataframe::
+            >>> rescaled_entry.df
 
         """
+        from copy import deepcopy
         from astropy import units as u
 
-        if yunit is None:
-            if self.y() == 'j':
-                yunit = u.A / u.m**2
-            elif self.y() == 'I':
-                yunit = u.A
-            else:
-                raise NotImplementedError("Unexpected naming of y axis.")
-
-        if yunit == 'original':
-            yunit = self.figure_description.axes.j.unit or self.figure_description.axes.I.unit
-
-        return u.Unit(yunit)
-
-    def df(self, xunit=None, yunit=None):
+        package = deepcopy(self.package)
+        fields = package.descriptor['resources'][0]['schema']['fields']
+        df = self.df.copy()
+
+        for idx, field in enumerate(fields):
+            if field['name'] in new_units.keys():
+                df[field['name']] *= u.Unit(field['unit']).to(u.Unit(new_units[field['name']]))
+                package.descriptor['resources'][0]['schema']['fields'][idx]['unit'] = new_units[field['name']]
+                package.descriptor['data description']['fields'][idx]['unit'] = new_units[field['name']]
+
+        import tempfile
+        import os
+
+        # with tempfile.TemporaryDirectory() as _tmpdir:
+        #     outdir=os.path.join(_tmpdir, package.descriptor['resources'][0]['path'])
+        #     print(outdir)
+        #     df.to_csv(outdir)
+        #     package.descriptor['resources'][0]['path'] = outdir
+        #     #package.infer([self.package outdir)
+        #     from datapackage import Package
+            
+        #     return Entry(package=Package(package.to_dict(), unsafe=True), bibliography=self.bibliography)
+
+        _tmpdir = tempfile.TemporaryDirectory()
+        outdir=os.path.join(_tmpdir.name, package.descriptor['resources'][0]['path'])
+        # print(outdir)
+        df.to_csv(outdir)
+        package.descriptor['resources'][0]['path'] = outdir
+        # package.infer([self.package outdir)
+        from datapackage import Package
+        
+        return Entry(package=Package(package.to_dict(), unsafe=True), bibliography=self.bibliography)
```

Comment:
@saraedum I could need some assistance here. 
With rescaled axes we have to save a rescaled dataframe as CSV.
The path to that file is stored in the resources section of the rescaled entry. Creating a dataframe from that temporary file is however not possible, since I guess, the file does not exist anymore.

Another possibility would be to overwrite the existing CSV and JSON file and load a new datapackage, which is probably not desired.

Also `package.to_dict()` seems to be deprecated. Haven't found a solution yet or if there is a new function. As mentioned it exists in frictionless.

Reply:
Has anybody ever looked into frictionless? Maybe we want to use that instead anyway?

Reply:
At least there, the resources of a data package can live in memory, see https://framework.frictionlessdata.io/docs/tutorials/schemes/buffer-tutorial. The same might be possible with our datapackage implementation.

Reply:
the trick with `entry._keepalive` seems to work.
I briefly try the buffering.